Sources/Generation/Generation.swift

```diff
@@ -22,88 +21,87 @@ public typealias InputTokens = [Int]
 public typealias GenerationOutput = [Int]
 
 /// A callable (a model, usually), that predicts the next token after a given sequence
-public typealias NextTokenModel = (InputTokens, GenerationConfig) -> any MLShapedArrayProtocol
+@available(macOS 15.0, iOS 18.0, *)
+public typealias NextTokenModel = (MLTensor, GenerationConfig) async -> MLTensor
 
 public typealias PredictionTokensCallback = (GenerationOutput) -> Void
 public typealias PredictionStringCallback = (String) -> Void
 
-// TODO: callbacks (for streaming)
+@available(macOS 15.0, iOS 18.0, *)
```

Comment:
This shouldn't be necessary as it's already specified in the package, but let's clear it later.